modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/java/com/liferay/frontend/data/set/admin/web/internal/model/DataSetModelManagerImpl.java

```diff
@@ -0,0 +1,1013 @@
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.frontend.data.set.admin.web.internal.model;
+
+import com.liferay.batch.engine.unit.BatchEngineUnitThreadLocal;
+import com.liferay.frontend.data.set.admin.model.DataSetModelManager;
+import com.liferay.object.constants.ObjectDefinitionConstants;
+import com.liferay.object.constants.ObjectFieldConstants;
+import com.liferay.object.constants.ObjectRelationshipConstants;
+import com.liferay.object.field.util.ObjectFieldUtil;
+import com.liferay.object.model.ObjectDefinition;
+import com.liferay.object.model.ObjectField;
+import com.liferay.object.model.ObjectFolder;
+import com.liferay.object.service.ObjectDefinitionLocalService;
+import com.liferay.object.service.ObjectFieldLocalService;
+import com.liferay.object.service.ObjectFolderLocalService;
+import com.liferay.object.service.ObjectRelationshipLocalService;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.dao.orm.QueryUtil;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.feature.flag.FeatureFlagManagerUtil;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.module.util.BundleUtil;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+import com.liferay.portal.vulcan.util.LocalizedMapUtil;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Daniel Sanz
+ */
+@Component(service = DataSetModelManager.class)
+public class DataSetModelManagerImpl implements DataSetModelManager {
+
+	public void checkCompany(long companyId) {
+		try {
+			if (_checkedCompanies == null) {
+				_checkedCompanies = new HashSet<>();
+			}
+
+			if (_checkedCompanies.contains(companyId)) {
+				return;
+			}
+
+			ObjectDefinition fdsViewObjectDefinition =
+				_objectDefinitionLocalService.fetchObjectDefinition(
+					companyId, "FDSView");
+
+			if (fdsViewObjectDefinition != null) {
+				_checkedCompanies.add(companyId);
+
+				return;
+			}
+
+			generate(_companyLocalService.getCompany(companyId));
+		}
+		catch (PortalException portalException) {
+			_log.error(portalException);
+		}
+	}
+
+	public void generate(Company company) {
```

Comment:
Does this need to be public? If not, it can probably be squashed with `checkCompany`.

Reply:
probably not, just `checkCompany()`  or whatever name we give to the entry point

Reply:
done